components/family-studies/api/src/main/java/org/phenotips/studies/family/internal/DefaultPedigree.java

```diff
@@ -108,6 +110,47 @@ public static boolean isSupportedPedigreeFormat(JSONObject data)
         return extractedObjects;
     }
 
+    @Override
+    public void addLink(String patientId)
+    {
+        int maxExistingPedigreeID = 0;
+
+        // go through all members to find first unused ID, and to check if the patient
+        // is already in the pedigree
+        JSONArray members = (JSONArray) this.data.opt(PEDIGREE_JSON_MEMBERS_KEY);
```

Comment:
`opt` is for when the value may be null. If this can indeed be null (unlikely, a pedigree should have at least one node), then check for null. If not, then it's simpler to use `getJSONArray` instead of `opt` + casting.